wolfcrypt/src/asn.c

```diff
@@ -22614,93 +22629,32 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
-             *
-             * load/receive check
-             *
-             * 1) Is CA boolean set?
-             *      No  - SKIP CHECK
-             *      Yes - Check key usage
-             * 2) Is Key usage extension present?
-             *      No  - goto 3
-             *      Yes - check keyCertSign assertion
-             *     2.a) Is keyCertSign asserted?
-             *          No  - goto 4
-             *          Yes - goto 3
-             * 3) Is pathLen set?
-             *      No  - goto 4
-             *      Yes - check pathLen against maxPathLen.
-             *      3.a) Is pathLen less than maxPathLen?
-             *           No - goto 4
-             *           Yes - set maxPathLen to pathLen and EXIT
-             * 4) Is maxPathLen > 0?
-             *      Yes - Reduce by 1
-             *      No  - ERROR
-             */
+        /* Set to WOLFSSL_MAX_PATH_LEN by default in InitDecodedCert_ex */
+        if (cert->pathLengthSet)
+            cert->maxPathLen = cert->pathLength;
 
-            if (cert->ca && cert->pathLengthSet) {
-                int checkPathLen = 0;
-                int decrementMaxPathLen = 0;
-                cert->maxPathLen = cert->pathLength;
-                if (cert->isCA) {
-                    WOLFSSL_MSG("\tCA boolean set");
-                    if (cert->extKeyUsageSet) {
-                        WOLFSSL_MSG("\tExtension Key Usage Set");
-                        if ((cert->extKeyUsage & KEYUSE_KEY_CERT_SIGN) != 0) {
-                            checkPathLen = 1;
-                        }
-                        else {
-                            decrementMaxPathLen = 1;
-                        }
-                    }
-                    else {
-                        checkPathLen = 1;
-                    } /* !cert->ca check */
-                } /* cert is not a CA (assuming entity cert) */
-
-                if (checkPathLen && cert->pathLengthSet) {
-                    if (cert->pathLength < cert->ca->maxPathLen) {
-                        WOLFSSL_MSG("\tmaxPathLen status: set to pathLength");
-                        cert->maxPathLen = cert->pathLength;
-                    }
-                    else {
-                        decrementMaxPathLen = 1;
-                    }
-                }
-
-                if (decrementMaxPathLen && cert->ca->maxPathLen > 0) {
-                    WOLFSSL_MSG("\tmaxPathLen status: reduce by 1");
-                    cert->maxPathLen = (byte)(cert->ca->maxPathLen - 1);
-                    if (verify != NO_VERIFY && type != CA_TYPE &&
-                                                    type != TRUSTED_PEER_TYPE) {
-                        WOLFSSL_MSG("\tmaxPathLen status: OK");
-                    }
-                } else if (decrementMaxPathLen && cert->ca->maxPathLen == 0) {
+        if (!cert->selfSigned) {
+            if (/* Need to perform a pathlen check on anything that will be used
```

Comment:
Don't mess the if statement up with comments.
Move the comment to above if statement.

Reply:
Moving

Reply:
Moved comments.